### DISCUSSION TITLE
Log full output not just training progress.

### DISCUSSION THREAD

**Discussion by lorinczszabolcs** [open-mmlab/mmselfsup#529] [Q&A] [answered]
**Describe the feature**
Logging for me only shows the training progress (loss values and eval as a function of training steps).

**Motivation**

1. It would be nice to keep the full standard output (including weight initialization), to be able to check if it has been done correctly. Many times the output is too long and can not scroll up enough to see if e.g. the backbone is initialized correctly, due to constraint in how much one is able to scroll up in the terminal.

**Additional context**
The output.log only contains logs from the start of the training, as shown in the picture:
![image](https://user-images.githubusercontent.com/13637835/183915477-7c2a7be8-0b3d-440e-a72a-0fa82f6ffbc0.png)

It would be really great if additional logging could be done (e.g. including the contents from the standard output right from the start, but I'm not sure how that could be solved. This feature would also be useful in other openmmlab libraries.

I would be open to make a PR if you could guide me in a direction where that could be solved / a preferred solution that would be optimal for you.

**Comment by fangyixiao18** [comment]
Hello, thanks for your reporting. 
Could you find a file called 'train_xxxx_xxxx', maybe like 'train_20220524_143504.log', here we have output the more information. 
e.g.
![1BgO5n4EvS](https://user-images.githubusercontent.com/36138628/184053026-cc23ce37-edb4-4c6a-b4fa-bddcd4af6842.jpg)

**Comment by lorinczszabolcs** [comment]
Hi! Thank you for your quick reply!

I only have a log called 'xxxx_xxxx.log.json' in `work_dir`, e.g. '20220810_104212.log.json', and it contains the following:

![image](https://user-images.githubusercontent.com/13637835/184071112-520c98c3-f0f7-444b-b24a-fe42c285f62c.png)

My wandb log folder has the following structure, but none of the files contain anything similar:
![image](https://user-images.githubusercontent.com/13637835/184073165-518c3158-adc4-41c2-9c12-ebd65ebd411d.png)

I also have a file called `INFO`, but that also only contains hooks and then the training progress:
![image](https://user-images.githubusercontent.com/13637835/184073397-fd2c42f4-d1a7-436d-8624-240f5f6afc4f.png)


No ''train_xxxx_xxxx.log' can be found in either `work_dir` or the wandb log folder unfortunately. Could it happen that my log config is somehow not correct?

**Comment by fangyixiao18** [comment]
could you provide your mmcv version and your command to start training? 
`.log` file is supposed to be in the work_dir folder

**Comment by lorinczszabolcs** [comment]
MMCV version is 1.6.1.

Training is done by a custom script, but the main steps are:
```python
...
# parse some args
...

# load config
cfg = Config.fromfile("...")
...

# create work_dir
cfg.work_dir = "../logs/{}".format(setting)

# set random seed
cfg.seed = 0
set_random_seed(0, deterministic=False)

# create the work directory
mmcv.mkdir_or_exist(osp.abspath(cfg.work_dir))

# set device and gpu
cfg.device = "cuda"
cfg.gpu_ids = range(num_gpus)
# build the algorithm
model = build_algorithm(cfg.model)

# init weights
model.init_weights()

# build the dataset
datasets = [build_dataset(cfg.data.train)]

# start pre-train
train_model(
    model,
    datasets,
    cfg,
    distributed=True if num_gpus > 1 else False,
    timestamp=time.strftime("%Y%m%d_%H%M%S", time.localtime()),
    meta=dict(),
)
```

**Comment by lorinczszabolcs** [comment]
Additional info: I am using mmsegmentation as well in another project, and the same issue appears there, only these logs are present in the mentioned directories, nothing similar to what you showed me. Could there be some kind of environment related issue? Both projects are running in docker containers.

**Comment by fangyixiao18** [comment]
> Additional info: I am using mmsegmentation as well in another project, and the same issue appears there, only these logs are present in the mentioned directories, nothing similar to what you showed me. Could there be some kind of environment related issue? Both projects are running in docker containers.

How did you write your `model`? Did your inherit from BaseModule from `mmcv`? https://github.com/open-mmlab/mmcv/blob/master/mmcv/runner/base_module.py#L15

**Comment by lorinczszabolcs** [comment]
My model inherits from `DenseCL` https://github.com/open-mmlab/mmselfsup/blob/8c4f76d513d67723a2004502357266c100e686a8/mmselfsup/models/algorithms/densecl.py#L12, so it should from `BaseModule` as well. In MMSegmentation my models inherit from `BaseSegmentor` 
https://github.com/open-mmlab/mmsegmentation/blob/dd42fa8d0125632371a41a87c20485494c973535/mmseg/models/segmentors/base.py#L13, so they also should inherit from `BaseModule` as well.

**Comment by fangyixiao18** [comment, marked as answer]
you can try initializing logger before `build_model`, like this https://github.com/open-mmlab/mmselfsup/blob/master/tools/train.py#L150

**Comment by lorinczszabolcs** [comment]
That solved the problem! Thanks!!
